qiskit/algorithms/evolvers/variational/solvers/ode/forward_euler_solver.py

```diff
@@ -0,0 +1,73 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Forward Euler ODE solver."""
+
+from typing import Iterable
+
+import numpy as np
+from scipy.integrate import OdeSolver
+from scipy.integrate._ivp.base import ConstantDenseOutput
+
+
+class ForwardEulerSolver(OdeSolver):
+    """Forward Euler ODE solver."""
+
+    def __init__(
+        self,
+        function: callable,
+        t0: float,
+        y0: Iterable,
```

Comment:
In the OdeSolver interface y0 is defined as array-like - which to me implies random access unlike an Iterable. Somehow I might have thought Sequence here to be more like array-like. I guess I was looking since the args really do not say much - just 'Initial state'

Reply:
It makes more sense to use the `Sequence`. Thanks!